### PR TITLE
[SYCL][ESIMD] Remove deprecated API

### PR DESCRIPTION
**Pull request by sndmitriev**
This patch removes deprecated API from ESIMD headers and corrects
simd_view::replicate APIs to be consistent with simd::replicate.

An associated llvm-test-suite change is https://github.com/intel/llvm-test-suite/pull/772

Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp [hunk 1]
```diff
@@ -463,7 +434,7 @@ class simd_view_impl {
   /// \return replicated simd instance.
   template <int Rep, int VS, int W, int HS>
   get_simd_t<element_type, Rep * W> replicate(uint16_t OffsetX) {
```
Comment (sndmitriev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:436]: @kbobrovs, should simd_view::replicate methods naming be consistent with simd::replicate?
Also, it looks like simd::replicate method called on line 413 does not exist.

File: sycl/include/sycl/ext/intel/experimental/esimd/common.hpp [hunk 2]
```diff
@@ -57,15 +57,6 @@
 
 // Macros for internal use
 #define __ESIMD_NS sycl::ext::intel::experimental::esimd
-#define __ESIMD_QUOTE1(m) #m
-#define __ESIMD_QUOTE(m) __ESIMD_QUOTE1(m)
-#define __ESIMD_NS_QUOTED __ESIMD_QUOTE(__ESIMD_NS)
-#define __ESIMD_DEPRECATED(new_api)                                            \
-  __SYCL_DEPRECATED("use " __ESIMD_NS_QUOTED "::" __ESIMD_QUOTE(new_api))
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/common.hpp:64]: I suggest to leave this one and above in place, as we'll likely have to deprecate APIs in future.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/elem_type_traits.hpp [hunk 3]
```diff
@@ -148,16 +148,6 @@ struct element_type_traits<T, std::enable_if_t<is_vectorizable_v<T>>> {
   static inline constexpr bool is_floating_point = std::is_floating_point_v<T>;
 };
 
-#ifdef __SYCL_DEVICE_ONLY__
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/elem_type_traits.hpp:151]: please verify that this change does not break internal builds.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp (hunk 1 above)
Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:436]: yes, they should, thanks for catching. could you please fix that too?

File: sycl/include/sycl/ext/intel/experimental/esimd/common.hpp (hunk 2 above)
Comment (sndmitriev) [reply sycl/include/sycl/ext/intel/experimental/esimd/common.hpp:64]: Ok.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp (hunk 1 above)
Comment (sndmitriev) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:436]: As part of this PR? Or in a separate PR?

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:436]: either way is OK for me

**Comment by sndmitriev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/772

**Review by kbobrovs — DISMISSED** [review]

**Comment by sndmitriev** [comment]
The following 9 test are expected to fail on existing llvm-test-suite. These tests will start passing once https://github.com/intel/llvm-test-suite/pull/772 is merged.

```
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/PrefixSum.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/Prefix_Local_sum3.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/histogram_raw_send.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/matrix_transpose.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/matrix_transpose2.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/matrix_transpose_glb.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/matrix_transpose_usm.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/regression/big_const_initializer.cpp
[2022-01-28T10:17:37.038Z]   SYCL :: ESIMD/regression/dgetrf_8x8.cpp
```

The following failures are unrelated to this patch

Jenkins/llvm-test-suite

LLVM_Test_Suite_Win, PI_Level_ZERO Test Step

```
[2022-01-28T12:41:50.108Z] Failed Tests (7):
[2022-01-28T12:41:50.108Z]   SYCL :: Basic/image/image_accessor_readsampler.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Basic/image/image_sample.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Sampler/normalized-clampedge-linear-float.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Sampler/normalized-clampedge-nearest.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Sampler/unnormalized-clamp-nearest.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Sampler/unnormalized-clampedge-linear-float.cpp
[2022-01-28T12:41:50.108Z]   SYCL :: Sampler/unnormalized-clampedge-nearest.cpp
```

I'll update this comment once all testing is finished if there will be any other unrelated failures.

**Review by kbobrovs — APPROVED** [review]

**Comment by kbobrovs** [comment]
According to @sndmitriev comments, the PR is good to go despite failures, merging.

**Pull request merged**